frontend/src/views/InvoiceView.vue

```diff
@@ -0,0 +1,243 @@
+<script setup lang="ts">
+import Button from "primevue/button";
+import Calendar from "primevue/calendar";
+import Card from "primevue/card";
+import Dropdown from "primevue/dropdown";
+import FloatLabel from "primevue/floatlabel";
+import SplitButton from "primevue/splitbutton";
+import Textarea from "primevue/textarea";
+import { useToast } from "primevue/usetoast";
+import { computed, onMounted, ref, watch } from "vue";
+import { useRoute } from "vue-router";
+
+import { getOrder } from "@/backendClient";
+import SubOrderItem from "@/components/orders/SubOrderItem.vue";
+import { ArticleKind } from "@/global/types/appTypes";
+import { PaymentStatus } from "@/global/types/appTypes";
+import type { InvoiceItemCreate } from "@/global/types/dataEditTypes";
+import type { Order } from "@/global/types/entities";
+import { calculateItemSumPrice, formatDateToIsoString } from "@/helpers/utils";
+import { ROUTES } from "@/router";
+
+const route = useRoute();
+const toast = useToast();
+
+let itemCount = 1;
+let calendarCount = 1;
+
+let invoiceItemsArray = ref<InvoiceItemCreate[]>([]);
+
+let invoiceInfo = ref({
+  order_id: "",
+  status: PaymentStatus.initial,
+  description: "",
+  invoice_date: "",
+  payment_target: "",
+  items: [],
+});
+
+let orderInfo = ref<Order | undefined>();
+const invoiceType = Object.values(PaymentStatus);
+let invoiceDate = ref<Date>();
+let paymentTarget = ref<Date>();
+let serviceDates = ref<{ id: number; date?: Date }[]>([{ id: 0 }]);
```

Comment:
why do service dates have an ID?

Reply:
Why don't we use the date (which shouldn't be optional) for the v-for key and the removal method?

Reply:
Because date could have no value when it just created

Reply:
Makes sense. Maybe we could still get rid of the IDs but let's keep it like this for now